ci/scripts/cpp_build.sh

```diff
@@ -31,11 +31,19 @@ elif [ -x "$(command -v xcrun)" ]; then
   export ARROW_GANDIVA_PC_CXX_FLAGS="-isysroot;$(xcrun --show-sdk-path)"
 fi
 
-# export CCACHE_COMPRESS=1
-# export CCACHE_COMPRESSLEVEL=5
-# export CCACHE_COMPILERCHECK=content
-# export CCACHE_DIR=/build/ccache
-# export PATH=/usr/lib/ccache/:$PATH
+export ARROW_USE_CCACHE=${ARROW_USE_CCACHE:-ON}
+
+if [ "${ARROW_USE_CCACHE}" == "ON" ]; then
+    export CCACHE_COMPILERCHECK=content
+    export CCACHE_COMPRESS=1
+    export CCACHE_COMPRESSLEVEL=5
+    export CCACHE_DIR=/build/ccache
+    export CCACHE_MAXSIZE=500M
+    export PATH=/usr/lib/ccache/:$PATH
```

Comment:
@pitrou Is it the same in case of conda as well?

Reply:
No. The conda version doesn't provide the symlinks that the Linux system version does. But at worse this line doesn't do anything.